### PR TITLE
GHDL abstracted with CLIAbstraction layer.

### PR DESCRIPTION
**Pull request by Paebbels**
# New Features
* Added editorconfig
* Added pyCharm project files.
* Enabled GHA workflow based on pyTooling/Actions.
* Added a (partial) program description of GHDL based on pyTooling.CLIAbstraction.
* Added a (partial) program description of Docker based on pyTooling.CLIAbstraction.
* Added a merged variant of "GHDL in Docker"
* Added a matching simple test case for the GHDL program class.
* Added setup.py for packaging.

# Changes
* Use `DEFAULT_CLASSIFIERS` in `setup.py`.
* Added option `-P<dir>` to GHDL.

# Bug Fixes
*None*

### CODE REVIEW

**Comment by umarcor** [comment]
@Paebbels, can you please have a look at the error during Packaging?

**Comment by Paebbels** [comment]
@umarcor it reports: `FileNotFoundError: [Errno 2] No such file or directory: 'pyEDAA/CLIAbstraction/__init__.py'`

I'll investigate it...

File: setup.py [hunk 1]
```diff
@@ -1,98 +1,52 @@
...
-
-	packages=setuptools_find_namespace_packages(exclude=["tests", "tests.*",]),
+"""\
+Package installer for 'An abstraction layer of EDA CLI tools'.
+"""
+from pathlib             import Path
+from pyTooling.Packaging import DescribePythonPackageHostedOnGitHub
+
+gitHubNamespace =        "edaa-org"
+packageName =            "pyEDAA.CLIAbstraction"
```
Comment (umarcor) [inline setup.py:39]: Here it is! This repo is `CLITool`. The one in pyTooling is `CLIAbstraction`.

File: setup.py [hunk 2]
```diff
@@ -1,98 +1,52 @@
...
+packageDirectory =       packageName.replace(".", "/")
+packageInformationFile = Path(f"{packageDirectory}/__init__.py")
+
+DescribePythonPackageHostedOnGitHub(
+	packageName=packageName,
+	description="An abstraction layer of EDA CLI tools.",
+	gitHubNamespace=gitHubNamespace,
+	sourceFileWithVersion=packageInformationFile,
+	developmentStatus="pre-alpha",
 	classifiers=[
```
Comment (umarcor) [inline setup.py:49]: We can copy https://github.com/edaa-org/pySVModel/blob/main/setup.py#L47

File: pyEDAA/CLITool/GHDL.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+#                                                                                                                      #
+# SPDX-License-Identifier: Apache-2.0                                                                                  #
+# ==================================================================================================================== #
+#
+"""\
+This module contains the CLI abstraction layer for `GHDL <https://github.com/ghdl/ghdl>`__.
+"""
+from pyTooling.CLIAbstraction            import CLIOption
+from pyTooling.CLIAbstraction.Executable import Executable
+from pyTooling.CLIAbstraction.Argument   import CommandArgument, ShortFlagArgument, LongFlagArgument, LongValuedFlagArgument
```
Comment (umarcor) [inline pyEDAA/CLITool/GHDL.py:38]: Maybe woth using multiple lines? `... import ( \n ... )`

File: setup.py (hunk 1 above)
Comment (Paebbels) [reply setup.py:39]: Will be fixed in 32d6fe6b.

File: pyEDAA/CLITool/GHDL.py (hunk 3 above)
Comment (Paebbels) [reply pyEDAA/CLITool/GHDL.py]: Done in 32688b10.

File: setup.py (hunk 2 above)
Comment (Paebbels) [reply setup.py:49]: Done in 801a1521.

**Comment by Paebbels** [comment]
@umarcor Merging with CI problems into `dev` so we can investigate CI solutions on a new branch to `dev`.

File: tests/unit/GHDL.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+from pytest               import mark
+from unittest             import TestCase
+
+from pyEDAA.CLITool.GHDL  import GHDL
+from .                    import Helper
+
+
+@mark.xfail      # XXX: workaround for problems in GHA
+class CommonOptions(TestCase, Helper):
+	_binaryDirectoryPath = Path(os_environ["GHDL_PREFIX"]) / "bin"
```
Comment (umarcor) [inline tests/unit/GHDL.py:43]: I believe this should be `getenv`, defaulting to `/usr/local/`. Moreover, `_binaryDirectoryPath` might be empty if GHDL_PREFIX is unset. E.g., if ghdl is available in the PATH already.

Comment (Paebbels) [reply tests/unit/GHDL.py:43]: Does the GHDL setup action add GHDL in path?

Comment (Paebbels) [reply tests/unit/GHDL.py:43]: Changed in 8046ed1a.

**Pull request merged**